skiko/src/macosMain/kotlin/org/jetbrains/skiko/redrawer/MetalRedrawer.macos.kt

```diff
@@ -115,17 +140,30 @@ internal class MacOsMetalRedrawer(
      */
     override fun redrawImmediately() {
         check(!isDisposed) { "MetalRedrawer is disposed" }
-        draw()
+        autoreleasepool {
+            if (!isDisposed) {
+                skiaLayer.update(getTimeNanos())
+                contextHandler.draw()
+            }
+        }
     }
 
-    private fun draw() {
-        // TODO: maybe make flush async as in JVM version.
+    private suspend fun draw() {
         autoreleasepool {
             if (!isDisposed) {
                 skiaLayer.update(getTimeNanos())
                 contextHandler.draw()
             }
         }
+
+        // When window is not visible - it doesn't make sense to redraw fast to avoid battery drain.
+        if (isWindowOccluded) {
+            withTimeoutOrNull(300) {
```

Comment:
I copied this 300 millisecond value from the JVM code to keep it consistent. However, I don't understand the reason for this delay, why not just suspend indefinitely (on native & jvm)? I can remove this timeout if needed.

Reply:
After some testing it seems all works fine without the 300ms timeout. CPU usage is reduced even more, on both native and JVM. Added a screenrecording with results for native app in the PR. I could not find any specific reason this timeout was added in git blame, so I removed it. If there's something I am missing please let me know.

What still stands out to me is that CPU and memory usage are both much lower for native vs JVM on my M1 Max macbook, with the same FPS result (100fps, my refresh rate). This is the also the case for my own much heavier app.